core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -786,4 +798,8 @@ int numPendingDeletes(Index index) {
             return deleteList.size();
         }
     }
+
+    public QueryParseContext newQueryParserContext() {
```

Comment:
What we end up doing in the REST layer at this point is creating it manually, have a look at RestSearchAction for instance. Not sure if we still want to have some kind of factory for it or not.